astropy/coordinates/tests/test_solar_system.py

```diff
@@ -417,3 +419,16 @@ def test_file_ephemeris_wrong_input():
     # Try loading a file that does exist, but is not an ephemeris file:
     with pytest.raises(ValueError):
         get_body('earth', time, ephemeris=__file__)
+
+
+def test_regression_10271():
+    t = Time(58973.534052125986, format='mjd')
+    # GCRS position of ALMA at this time
+    obs_p = CartesianRepresentation(5724535.74068625, -1311071.58985697, -2492738.93017009, u.m)
+
+    geocentre = CartesianRepresentation(0, 0, 0, u.m)
+    icrs_sun_from_alma = _get_apparent_body_position('sun', t, 'builtin', obs_p)
+    icrs_sun_from_geocentre = _get_apparent_body_position('sun', t, 'builtin', geocentre)
+
+    difference = (icrs_sun_from_alma - icrs_sun_from_geocentre).norm()
+    assert_quantity_allclose(difference, 0.13046941*u.m)
```

Comment:
Ah, I just noticed the CI 32-bit failure is real - if you put, say `atol=1*u.mm` here, it should be OK.